### PR TITLE
summoner: provide phase 1 root during ceremony initialization

### LINKED ISSUE #3156: Summonerd: Initializing phase 1 root
We want to be able to at least setup the start of phase1 by using a "blank" set of parameters, which are still valid, just deterministically generated.

This command can later be modified to read some kind of file with a desired starting output.

### PR DESCRIPTION
**Pull request by redshiftzero**
Closes #3156

Note that I have intentionally not done the transition from the provided phase 1 to phase 2, which will be done in #3158 which I'll pick up after this one.

### CODE REVIEW

**Review by cronokirby — CHANGES_REQUESTED** [review]
My thinking was that instead initialization should be a separate command, rather than something done automatically.

The reason behind that is that then we could make this command take in a non-dummy root from an external source to populate things instead.

**Comment by redshiftzero** [comment]
The commands we have now are:

```
Running `target/debug/summonerd --help`
summonerd v0.62.0-11-g6d156c5
Penumbra summoning ceremony coordinator

USAGE:
    summonerd [OPTIONS] <SUBCOMMAND>

OPTIONS:
    -h, --help                             Print help information
        --tokio-console <tokio-console>    Enable Tokio Console support [default: false]
    -V, --version                          Print version information

SUBCOMMANDS:
    generate-phase1    Generate a phase 1 root (for testing purposes)
    help               Print this message or the help of the given subcommand(s)
    init               Initialize the coordinator
    start              Start the coordinator
```

I tested this via:

1. Generate a test phase 1 root and store in a file (if we use an externally provided phase 1 root later we can delete this): 
```
cargo run --bin summonerd -- generate-phase1 --output testphase1.txt
```

2. Initialize the database using the provided phase 1 root:
```
cargo run --bin summonerd -- init --storage-dir /tmp/summonerd --phase1-root testphase1.txt
```

After this step the database exists and we can validate that the phase 1 root has been inserted in the database:

```
% sqlite3 /tmp/summonerd/ceremony.db
SQLite version 3.39.5 2022-10-14 20:58:05
Enter ".help" for usage hints.
sqlite> sqlite> select * from phase1_contributions;
0|1|������n���N9��H�\�7�Ȃ��A��,��T���%b��g
t��H�|
```

3. Now we start the summonerd server as before:

```
cargo run --bin summonerd -- start --storage-dir /tmp/summonerd --fvk YOURFVKHERE --node https://grpc.testnet-preview.penumbra.zone
```

File: tools/summonerd/src/main.rs [hunk 1]
```diff
@@ -87,6 +105,40 @@ impl Opt {
...
+                let mut phase_1_bytes = Vec::new();
+                let mut buffer = [0; 4096 * 10]; // 40 KB chunks
+
+                loop {
+                    let bytes_read = reader.read(&mut buffer)?;
+                    if bytes_read == 0 {
+                        break;
+                    }
+                    phase_1_bytes.extend_from_slice(&buffer[..bytes_read]);
+                }
```
Comment (cronokirby) [inline tools/summonerd/src/main.rs:124]:
```suggestion
                reader.read_to_end(&mut phase_1_bytes)?;
```

I believe this should work

Comment (cronokirby) [reply tools/summonerd/src/main.rs:124]: (We're already doing buffered reads via BufReader)

**Review by cronokirby — APPROVED** [review]
LGTM, I'll just apply my suggestion, check CI, and then merge

**Pull request merged**